parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -273,7 +273,11 @@ fi
 {{end}}
 
 {{- /* re-enable unattended upgrades */}}
+{{- if EnableUnattendedUpgrades}}
 rm -f /etc/apt/apt.conf.d/99periodic
+{{else}}
+apt_get_purge unattended-upgrades || exit {{GetCSEErrorCode "ERR_APT_PURGE_FAIL"}}
```

Comment:
Why purge the unattended-upgrades?

It would be better to leave the code in case someone wants to manage this themselves as otherwise they will have to get the code some other way or get it and accidentally enable the automatic processes.

The better process is to set the apt configuration to turn off `APT::Periodic::Unattended-Upgrade`

It is set to "1" by default in this distro setup.  Setting that to "0" disables the automatic process but leaves the code there such that a manual process can be executed.

Note that there is also the apt database update which can also be turned off via `APT::Periodic::Update-Package-Lists`

Note that these are the defaults in 18.04:
```
# apt-config dump | grep Periodic
APT::Periodic "";
APT::Periodic::Update-Package-Lists "1";
APT::Periodic::Download-Upgradeable-Packages "0";
APT::Periodic::AutocleanInterval "0";
APT::Periodic::Unattended-Upgrade "1";
```

It would be much better to address it this way than removing the code.

Reply:
PS - the file `/etc/apt/apt.conf.d/20auto-upgrades` is where this configuration lives in the ubuntu distro we use.

See:
```
# grep -r Periodic /etc/apt/apt.conf.d
/etc/apt/apt.conf.d/10periodic:APT::Periodic::Update-Package-Lists "1";
/etc/apt/apt.conf.d/10periodic:APT::Periodic::Download-Upgradeable-Packages "0";
/etc/apt/apt.conf.d/10periodic:APT::Periodic::AutocleanInterval "0";
/etc/apt/apt.conf.d/20auto-upgrades:APT::Periodic::Update-Package-Lists "1";
/etc/apt/apt.conf.d/20auto-upgrades:APT::Periodic::Unattended-Upgrade "1";
```

Reply:
Thanks for the feedback, this makes sense.

Rather than just try to intercept the default apt configuration we expect Ubuntu to pave, should we simply retain the cloud-init-delivered configuration, which runs at a higher priority (we name the file `/etc/apt/apt.conf.d/99periodic`) and disables all automatic background operations? See:

https://github.com/Azure/aks-engine/blob/master/parts/k8s/cloud-init/nodecustomdata.yml#L156

Reply:
That works - and avoids having to edit existing files.  It also cleanly separates control/option from code.